pkg/service/node.go

```diff
@@ -43,16 +43,20 @@ func (n *NodeService) NodeStageVolume(_ context.Context, req *csi.NodeStageVolum
 	// is there a filesystem on this device?
 	filesystem, err := getDeviceInfo(device)
 	if err != nil {
+		klog.Errorf("Failed to fetch device info for volume %s on node %s", req.VolumeId, n.nodeId)
 		return nil, err
 	}
+	if filesystem != "" {
+		klog.Infof("Detected fs %s, returning", filesystem)
+		return &csi.NodeStageVolumeResponse{}, nil
+	}
+
 	fsType := req.VolumeCapability.GetMount().FsType
-	if filesystem == "" {
-		// no filesystem - create it
-		klog.Infof("Creating FS %s on device %s", fsType, device)
-		makeFSErr := makeFS(device, fsType)
-		if makeFSErr != nil {
-			return nil, makeFSErr
-		}
+	// no filesystem - create it
+	klog.Infof("Creating FS %s on device %s", fsType, device)
+	makeFSErr := makeFS(device, fsType)
```

Comment:
can you name that erorr err as well? having too many names is a smell because here we immediatly return is there is an err.

Reply:
ack